qiskit_ibm_runtime/ibm_estimator.py

```diff
@@ -0,0 +1,141 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Qiskit Runtime Estimator primitive service."""
+
+from typing import Iterable, Optional, Dict
+
+from qiskit.circuit import QuantumCircuit, Parameter
+from qiskit.quantum_info import SparsePauliOp
+
+from .base_primitive import BasePrimitive
+from .estimator import Estimator
+
+
+class IBMEstimator(BasePrimitive):
+    """Class for interacting with Qiskit Runtime Estimator primitive service.
+
+    Qiskit Runtime Estimator primitive service estimates expectation values of quantum circuits and
+    observables.
+
+    IBMEstimator can be initialized with the following parameters. It returns a factory.
+
+    * service: Optional instance of :class:`qiskit_ibm_runtime.IBMRuntimeService` class,
+        defaults to `IBMRuntimeService()` which tries to initialize your default saved account.
+
+    * backend: Optional instance of :class:`qiskit_ibm_runtime.IBMBackend` class or
+        string name of backend, if not specified a backend will be selected automatically
+        (IBM Cloud only).
+
+    The factory can then be called with the following parameters to initialize the Estimator
+    primitive. It returns an :class:`qiskit_ibm_runtime.sessions.EstimatorSession` instance.
+
+    * circuits: list of (parameterized) quantum circuits
+        (a list of :class:`~qiskit.circuit.QuantumCircuit`))
+
+    * observables: a list of :class:`~qiskit.quantum_info.SparsePauliOp`
+
+    * parameters: a list of parameters of the quantum circuits.
+        (:class:`~qiskit.circuit.parametertable.ParameterView` or
+        a list of :class:`~qiskit.circuit.Parameter`) specifying the order
+        in which parameter values will be bound.
+
+    The :class:`qiskit_ibm_runtime.sessions.EstimatorSession` instance can be called repeatedly
+    with the following parameters to estimate expectation values.
+
+    * circuit_indices: A list of circuit indices.
+
+    * observable_indices: A list of observable indices.
+
+    * parameter_values: Concrete parameters to be bound.
+
+    All the above lists should be of the same length.
+
+    Example::
+
+        from qiskit.circuit.library import RealAmplitudes
+        from qiskit.quantum_info import SparsePauliOp
+
+        from qiskit_ibm_runtime import IBMRuntimeService, IBMEstimator
+
+        service = IBMRuntimeService(auth="cloud")
+        estimator_factory = IBMEstimator(service=service, backend="ibmq_qasm_simulator")
```

Comment:
I'll look later, but is `backend` always a string, or can it also be a backend object?

Reply:
It can also be a backend object.